### PR TITLE
Updates signing tests and documentation

### PR DESCRIPTION
**Pull request by alex-miller-0**
* Breaks signing tests into individual files
* Splits ETH tx and ABI test scripts
* Updates signing and testing documentation

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -507,9 +507,10 @@ function getFwVersionConst (v: Buffer): FirmwareConstants {
     c.eip712MaxTypeParams = 36;
   }
   // V0.14.1 Added the Terra decoder
```
Comment (douglance) [inline src/constants.ts:509]: Should we just remove this entirely?

File: docs/docs/signing.md [hunk 2]
```diff
@@ -79,84 +99,63 @@ const req = {
 const sig = await client.sign(req)
 ```
 
-### 💾 Calldata Decoding
+## 💾 Calldata Decoding
 
-:::note
-All available calldata decoding options will be documented in this section. More may be added as time goes on.
-:::
+> NOTE: All available calldata decoding options will be documented in this section. More may be added as time goes on.
```
Comment (douglance) [inline docs/docs/signing.md:104]: The previous note syntax: 

```
:::note
...
:::
```

Makes it look like these nice callouts in the docs: https://gridplus.github.io/gridplus-sdk/signing#-calldata-decoding

File: docs/docs/signing.md [hunk 3]
```diff
@@ -79,84 +99,63 @@ const req = {
...
 If you include a valid calldata decoder, the appearance of the transaction's data on the user's Lattice should transform from a raw hex string to a markdown-style version which displays the function name, parameter names, and values.
 
-#### Storing Calldata Decoders
+### 1️⃣ EVM Calldata Decoding
 
-Although not necessary, in certain situations it may be advantageous to pre-save decoders to the Lattice. One advantage is that if the decoder is saved, you do not need to include it in the transaction request, which frees up some space. Additionally, pre-saving data may unlock certain security features depending on the decoder type.
+EVM transactions serialize calldata according to the [Ethereum ABI specification](https://docs.soliditylang.org/en/latest/abi-spec.html). The first four bytes of a transaction's `data` represent the "function selector", which is (sort of) a unique identifier for a given function. 
 
-You can use the following API:
+> NOTE: We do not support 100% of all edge cases in the ABI specification, but we do support the vast majority of types.  Please open a pull request or an issue if your request fails to decode on a Lattice.
```
Comment (douglance) [inline docs/docs/signing.md:122]: We could use the `:::note` syntax here

File: docs/docs/signing.md [hunk 4]
```diff
@@ -79,84 +99,63 @@ const req = {
...
 
-* `addDecoders`: Allows the user to add a series of calldata decoders for a specific decoder type (e.g. EVM). This will prompt the user to approve these decoders on the target Lattice before returning success.
-* `getDecoders`: Fetch `n` consecutive decoders for a specific type, starting a specific index.
-* `removeDecoders`: Remove a set of included decoders for a specific type. You can also set a flag to remove all decoders for a specific type.
+> NOTE: `fetchCalldataDecoder` takes in params `(tx.input, tx.to, tx.chainId, shouldRecurse)`. The first 3 come from the transaction object (note that `chainId` must be a regular integer), while `shouldRecurse` is used to flag whether to look up nested definitions, as is typical with contract patterns like `multicall`. You can only use `shouldRecurse` with Lattice firmware v0.16.0 and above.
```
Comment (douglance) [inline docs/docs/signing.md:126]: :::note ? :smile:

File: docs/docs/signing.md [hunk 5]
```diff
@@ -221,13 +220,11 @@ const reqData = {
...
 
-#### `sign_typed_data`
+### `sign_typed_data`
 
 This is used in protocols such as EIP712. It is meant to be an encoding for JSON-like data that can be more human readable.
 
-:::note
-Only `sign_typed_data` V3 and V4 are supported.
-:::
+> NOTE: Only `sign_typed_data` V3 and V4 are supported.
```
Comment (douglance) [inline docs/docs/signing.md:227]: :::note

File: docs/docs/signing.md [hunk 6]
```diff
@@ -79,84 +99,63 @@ const req = {
...
+  payload: <Raw message to be signed, e.g. serialized transaction>,
+  decoder: <Optional serialized information about decoding the payload>
+}
 await client.sign(req);
 ```
 
 If you include a valid calldata decoder, the appearance of the transaction's data on the user's Lattice should transform from a raw hex string to a markdown-style version which displays the function name, parameter names, and values.
 
-#### Storing Calldata Decoders
+### 1️⃣ EVM Calldata Decoding
```
Comment (douglance) [inline docs/docs/signing.md:118]: Do we want to include any images of decoded data vs. non-decoded? We have some in progress.

File: src/__test__/e2e/signing/determinism.test.ts [hunk 7]
```diff
@@ -26,584 +24,585 @@ import {
   gpErrors,
   jobTypes,
 } from '../../utils/helpers';
-import { initializeSeed } from '../../utils/initializeClient';
+import { initializeClient, initializeSeed } from '../../utils/initializeClient';
 import { runTestCase } from '../../utils/runners';
 let seed: Buffer;
+let client;
```
Comment (douglance) [inline src/__test__/e2e/signing/determinism.test.ts:30]: I don't think we need to define `client` outside of the scope of the `describe` block, do we? Other tests just define it as a `const` within the `describe` block.

File: package.json [hunk 8]
```diff
@@ -25,7 +25,12 @@
     "e2e-gen": "vitest src/__test__/e2e/general.test.ts",
     "e2e-kv": "vitest src/__test__/e2e/kv.test.ts",
     "e2e-ne": "vitest src/__test__/e2e/non-exportable.test.ts",
-    "e2e-sign": "vitest src/__test__/e2e/signing/signing.test.ts",
+    "e2e-sign": "npm run e2e-sign-determinism && npm run e2e-sign-evm-abi && npm run e2e-sign-evm-tx && npm run e2e-sign-solana && npm run e2e-sign-unformatted",
```
Comment (douglance) [inline package.json:28]: `vitest src/__test__/e2e/signing` might accomplish the same result as calling all of the files individually (and it should be more future-proof)

**Review by douglance — COMMENTED** [review]
Looks good. Just had some minor notes.

File: docs/docs/signing.md (hunk 2 above)
Comment (alex-miller-0) [reply docs/docs/signing.md:104]: Oooh they weren't rendering on my markdown viewer so I was confused. Will revert

File: src/__test__/e2e/signing/determinism.test.ts (hunk 7 above)
Comment (alex-miller-0) [reply src/__test__/e2e/signing/determinism.test.ts:30]: Yeah you're right. Just didn't want to lose scope. Will change

File: src/constants.ts (hunk 1 above)
Comment (alex-miller-0) [reply src/constants.ts:509]: No I'd like to keep it as reference to document the enum values. Also we will likely re-use a lot of it if/when we do Cosmos

File: package.json (hunk 8 above)
Comment (alex-miller-0) [reply package.json]: [074aaa8](https://github.com/GridPlus/gridplus-sdk/pull/467/commits/074aaa820cb7ae45a3de58c7cbafbbc6c1fb8287)

File: docs/docs/signing.md (hunk 5 above)
Comment (alex-miller-0) [reply docs/docs/signing.md]: [074aaa8](https://github.com/GridPlus/gridplus-sdk/pull/467/commits/074aaa820cb7ae45a3de58c7cbafbbc6c1fb8287)

File: docs/docs/signing.md (hunk 4 above)
Comment (alex-miller-0) [reply docs/docs/signing.md]: [074aaa8](https://github.com/GridPlus/gridplus-sdk/pull/467/commits/074aaa820cb7ae45a3de58c7cbafbbc6c1fb8287)

File: docs/docs/signing.md (hunk 3 above)
Comment (alex-miller-0) [reply docs/docs/signing.md]: [074aaa8](https://github.com/GridPlus/gridplus-sdk/pull/467/commits/074aaa820cb7ae45a3de58c7cbafbbc6c1fb8287)

File: docs/docs/signing.md (hunk 2 above)
Comment (alex-miller-0) [reply docs/docs/signing.md]: [074aaa8](https://github.com/GridPlus/gridplus-sdk/pull/467/commits/074aaa820cb7ae45a3de58c7cbafbbc6c1fb8287)

File: docs/docs/signing.md (hunk 6 above)
Comment (douglance) [reply docs/docs/signing.md:120]: Looks like [that ticket](https://github.com/GridPlus/gridplus-io/issues/18) isn't done yet. I created [an issue](https://github.com/GridPlus/gridplus-sdk/issues/468) to update images in the sdk docs in the future.

**Review by douglance — APPROVED** [review]

**Pull request merged**